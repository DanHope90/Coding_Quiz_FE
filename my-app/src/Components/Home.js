import React from "react";

function Home(){
    return(<div>This is my home page!</div>)
}

export default Home;