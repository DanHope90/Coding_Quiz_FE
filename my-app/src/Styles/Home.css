.home-text {
    display: flex;
    flex-direction: column;
    gap: 50px;
    font-size: 20px;
}

.user-name {
    font-size: 22px;
}

