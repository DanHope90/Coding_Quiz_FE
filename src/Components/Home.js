import React, { useContext } from "react";
import Context from "../Utils/Context";
import "../Styles/Home.css";

function Home() {
  const { userInfo } = useContext(Context);

  return (
    <div className="home-text">
      <div>Welcome to Dev Quiz!</div>
      <div>You are logged in as: </div>
      <div className="user-name">{userInfo.userName}</div>
      <div>Use the navigation bar above to take or view your quizzes.</div>
    </div>
  );
}

export default Home;
