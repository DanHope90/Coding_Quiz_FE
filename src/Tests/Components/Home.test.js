import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import Home from '../../Components/Home';
import Context from "../../Utils/Context";

test("renders the Home component correctly", () => {
  const testRenderer = renderer.create(
    <Context.Provider value={jest.fn()}>
      <BrowserRouter>
        <Home />
      </BrowserRouter>
    </Context.Provider>,
  );
  expect(testRenderer).toMatchSnapshot();
});
