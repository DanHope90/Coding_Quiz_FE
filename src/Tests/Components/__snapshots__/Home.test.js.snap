// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Home component correctly 1`] = `
<div>
  This is my home page!
</div>
`;
